contracts/src/libraries/LPMath.sol

```diff
@@ -887,15 +904,23 @@ library LPMath {
 
             // NOTE: Round the delta down to avoid overshooting.
             //
-            // We calculate the updated share proceeds `x_n+1` by proceeding
-            // with Newton's method. This is given by:
-            //
-            // x_n+1 = x_n - F(x_n) / F'(x_n)
+            // Calculate the objective functions value. If the value's magnitude
+            // is greater than the previous iteration, we break out of the loop
+            // since Newton's method has begun to diverge.
             int256 delta = presentValue.mulDown(lpTotalSupply).toInt256() -
                 _params
                     .startingPresentValue
                     .mulUp(_params.activeLpTotalSupply)
                     .toInt256();
+            if (lastDelta != 0 && delta.abs() > lastDelta.abs()) {
+                break;
+            }
+            lastDelta = delta;
```

Comment:
This change is intended to improve our security assumptions so that people don't pass a huge `maxIterations` that leads to divergence and a larger LP share price deviation than we expect.